### PR TITLE
Updates to fix the Endpoints sample.

### PR DESCRIPTION
**Pull request by wmwong**
Updates gem versions.
Updates the README to mention usage of Ruby 2.0.0+ and to fix the deploy
command.
Fixes the echo client as to_json method doesn't exist.
Fixes the google id token client as 'open' not a command in Linux.

### CODE REVIEW

File: appengine/endpoints/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+endpoints/
```
Comment (remi) [inline appengine/endpoints/.gitignore]: Does an endpoints/ subdirectory get used or generated by this sample?  Just wondering why to ignore this subdirectory

File: appengine/endpoints/README.md [hunk 2]
```diff
@@ -1,7 +1,9 @@
 # Google Cloud Endpoints & App Engine Flexible Environment & Ruby
 
 This sample demonstrates how to use Google Cloud Endpoints on
-Google App Engine flexible environment using Ruby.
+Google App Engine flexible environment using Ruby. This sample requires that you
+have [Ruby](https://www.ruby-lang.org/en/documentation/installation/) 2.0.0 or
```
Comment (remi) [inline appengine/endpoints/README.md]: Out of curiosity, did you run into any problems running this sample with previous versions of Ruby?

The stable, maintained versions of Ruby include 2.3* 2.2* and 2.1*.  Ruby 2.0* reached end of life [earlier this year](https://www.ruby-lang.org/en/news/2016/02/24/support-plan-of-ruby-2-0-0-and-2-1/).

This comment is 👍  but I'll come back through and update all READMEs to mention that samples are only intended to run on 2.1+ (and update travis to run 2.3, 2.2, 2.1)

Mostly curious if this blew up for you! 😄  Lemme know

File: appengine/endpoints/README.md [hunk 3]
```diff
@@ -40,7 +42,7 @@ Open the `swagger.yaml` file and in the `host` property, replace
 
 Then, deploy the sample using `gcloud`:
 
-    gcloud app deploy app.yaml
+    gcloud beta app deploy
```
Comment (remi) [inline appengine/endpoints/README.md]: Nice!  Should this be `gcloud app deploy`, though?

Correct to remove app.yaml, but deploying Google App Engine flexible environment applications isn't under `beta` [[1]](https://cloud.google.com/sdk/gcloud/reference/app/deploy)

File: appengine/endpoints/clients/echo_client.rb [hunk 4]
```diff
@@ -49,7 +50,7 @@
 options[:message] = "Hello echo message" unless options[:message]
 
 url = "#{options[:host]}/echo?key=#{options[:api_key]}"
-body = { message: options[:message] }.to_json
+body = JSON.generate(message: options[:message])
```
Comment (remi) [inline appengine/endpoints/clients/echo_client.rb]: Is this change necessary with the `require "json"` added?

File: appengine/endpoints/clients/google_id_token_client.rb [hunk 5]
```diff
@@ -73,7 +73,8 @@
   redirect_uri:         "urn:ietf:wg:oauth:2.0:oob"
 )
 
-`open "#{oauth.authorization_uri}"`
```
Comment (remi) [inline appengine/endpoints/clients/google_id_token_client.rb]: Thank you for fixing this!  Not cross-platform compat at all!  😄

File: appengine/endpoints/.gitignore (hunk 1 above)
Comment (wmwong) [inline appengine/endpoints/.gitignore]: Yes. When doing "glcoud beta app deploy", endpoints/service.json gets created.

File: appengine/endpoints/README.md (hunk 2 above)
Comment (wmwong) [inline appengine/endpoints/README.md]: This blew up when using 1.9.3, which was installed by default on my Ubuntu computer.

File: appengine/endpoints/README.md (hunk 3 above)
Comment (wmwong) [inline appengine/endpoints/README.md]: For Endpoints, we need to use beta or the Endpoints related stuff doesn't happen.

File: appengine/endpoints/clients/echo_client.rb (hunk 4 above)
Comment (wmwong) [inline appengine/endpoints/clients/echo_client.rb]: Oh, my mistake. The documentation does mention #to_json:
http://ruby-doc.org/stdlib-2.0.0/libdoc/json/rdoc/JSON.html#module-JSON-label-Generating+JSON

I'll revert this.
